tests/test_relations_hyperlink.py

```diff
@@ -68,6 +69,22 @@ class Meta:
         fields = ('url', 'name', 'nullable_source')
 
 
+# Nullable OneToOne2
+class NullableOneToOneTargetSerializer2(serializers.HyperlinkedModelSerializer):
+    nullable_source2 = serializers.HyperlinkedRelatedField(
+        read_only=True,
+        view_name='nullableonetoonesource2-detail',
+        # This source causes a AttributeError, but seems like the proper value
+        # source='nullable_source.nullable_source2',
+        # This source produces the pk of the nullable_source instance, not nullable_source2
+        source='nullable_source',
```

Comment:
As [stated before](https://github.com/tomchristie/django-rest-framework/issues/2826#issuecomment-93620548), this is not how the `source` argument works. You shouldn't be expecting `OneToOneTarget.nullable_source` to give you anything other than `OneToOneTarget.nullable_source` (you're expecting `OneToOneTarget.nullable_source.nullable_source2`).

If there is an error in the code, it would happen when `source` is set to `nullable_source.nullable_source2`.